File: seeker/snippet/send.py
#date: 2024-05-10T17:05:40Z
#url: https://api.github.com/gists/986b033d315ee1f5e25ed6c7d2b50820
#owner: https://api.github.com/users/agatemosu

import asyncio

HOST = "127.0.0.1"
PORT = 62775


async def send_reboot():
    try:
        _, writer = await asyncio.open_connection(HOST, PORT)

        writer.write(b"reboot")
        await writer.drain()

        writer.close()
        await writer.wait_closed()

    except ConnectionError as error:
        print(error.strerror)


async def main():
    print("Sending reboot to", f"{HOST}:{PORT}")
    await send_reboot()


if __name__ == "__main__":
    asyncio.run(main())

File: seeker/snippet/send.go
//date: 2024-08-23T16:47:00Z
//url: https://api.github.com/gists/97370d1c5641ba0480494e75c2b73804
//owner: https://api.github.com/users/robin019

package main

import (
	amqp "github.com/rabbitmq/amqp091-go"
	"log"
)

func main() {
	conn, err := amqp.Dial("amqp://guest:guest@172.17.0.2:5672/")
	if err != nil {
		log.Fatalf("Failed to connect to RabbitMQ: %v", err)
	}
	defer conn.Close()

	ch, err := conn.Channel()
	if err != nil {
		log.Fatalf("Failed to open a channel: %v", err)
	}
	defer ch.Close()

	q, err := ch.QueueDeclare(
		"hello", // name
		true,    // durable
		false,   // delete when unused
		false,   // exclusive
		false,   // no-wait
		map[string]interface{}{
			"x-queue-type": "quorum",
		}, // arguments
	)
	if err != nil {
		log.Fatalf("Failed to declare a queue: %v", err)
	}

	body := "Hello World!"
	err = ch.Publish(
		"",     // exchange
		q.Name, // routing key
		false,  // mandatory
		false,  // immediate
		amqp.Publishing{
			ContentType: "text/plain",
			Body:        []byte(body),
		})
	if err != nil {
		log.Fatalf("Failed to publish a message: %v", err)
	}
	log.Printf(" [x] Sent %s\n", body)
}